scripts/oda_data.py

```diff
@@ -102,25 +102,47 @@ def read_gni():
         donor_name=lambda d: country_converter.convert(d.donor_name, to="short_name")
     )
 
-
-def read_idrc_estimates():
+def read_idrc_estimates() -> pd.DataFrame:
     """Read our estimates from the GoogleSheet which summarises the analysis"""
     url = (
         "https://docs.google.com/spreadsheets/d/e/2PACX-1vSqAIxjSZ78fE93CP1K9K0t8rL"
         "M2wi0z_nc60ezrUeDEIOPz-vr01SmmS_5nNnq_uPE0dM26m0V3rQK/pub?gid=1887834724"
         "&single=true&output=csv"
     )
     return (
-        pd.read_csv(url)
-        .assign(
-            donor_name=lambda d: country_converter.convert(d.iso_code, to="short_name"),
-            year=2022,
-        )
-        .rename(columns={"estimated_idrc": "idrc"})
-        .drop("iso_code", axis=1)
+        df=pd.read_csv(url).assign(donor_name=lambda d: country_converter.convert(d.iso_code, to="short_name"),
+            year=2022).rename(columns={"estimated_idrc": "idrc"}).drop("iso_code", axis=1)
     )
 
 
+def read_idrc_annual_ratios() -> pd.DataFrame:
+    """Read our estimates from the CSV which provides ratios of IDRC expenditure for 2022, 2023 and 2024"""
+
+    return (
+        pd.read_csv(f"{config.PATHS.output}/hcr_data.csv").rename(columns={"Country": "donor_name"}).assign(donor_name=lambda d: country_converter.convert(d.iso_code, to="short_name"))
+            )
+
+
+def estimate_annual_idrc(df: pd.DataFrame) -> df.DataFrame:
+    """Calculate estimates for annual idrc. Multiply average IDRC by difference and ratio for each month"""
+
+    # Merge DataFrames for calculations (Q: does this break single responsibility? Feels excessive to have this as its own function)
```

Comment:
I mean we could be very extreme with single responsibility. But not really since the purpose of this is to create a pipeline for that analysis. So you need to bring different data together, etc